docs/odh/fetch-upstream-release.md

```diff
@@ -0,0 +1,430 @@
+This repository is a fork of [kserve/kserve](https://github.com/kserve/kserve)
+repository to integrate with Open Data Hub platform. The releases on ODH
+platform have a faster cadence than the upstream repository. Still, an upstream
+stable release is used as a base for any releases of the ODH Project. We
+accomplish this by creating a branch in ODH fork based from an upstream release
+tag. Any integration changes to ODH Platform are added on top of the created
+branch.
+
+This guide outlines the process to fetch a stable upstream release to ODH fork.
+
+# Prerequisites
+
+As you will be are creating a _new_ branch, the configured branch protections
+may not apply to it. Despite this, you must still adhere to [GitHub
```

Comment:
we need to manually add the new branch under the protected ones, no?

Reply:
I reconfigured the branch protections so that it catches all branches named `release-v*`. So, even when new, the branch protections should apply and no action should be expected.

I would leave this is just as a note in case configuration changes.

> we need to manually add the new branch under the protected ones, no?

But to answer your question, yes they need to be reviewed and manually added. There is a mention of this at the very end of the guide.